### DISCUSSION TITLE
Conditional logging per model?

### DISCUSSION THREAD

**Discussion by wiegand** [palkan/logidze#248] [Q&A] [answered]
Does Logidze offer the ability to conditionally enable logging in a similar manner to [how the Audited gem does?](https://github.com/collectiveidea/audited?tab=readme-ov-file#conditional-auditing):

```ruby
class User < ActiveRecord::Base
  audited if: :active?

  def active?
    last_login > 6.months.ago
  end
end
```

For my specific case, I don't need the conditional logic to be in the app. My model is using Rails's STI, so I have a `type` column which, if it matches a particular string, I would like to NOT log its data. Thank you!

**Comment by palkan** [comment, marked as answer]
Since Logidze operates at the database-level, we cannot provide a similar model-backed feature. In your case, you may consider changing the trigger code a bit to include the type filter:

```sql
CREATE TRIGGER logidze_on_posts
BEFORE UPDATE OR INSERT ON posts FOR EACH ROW
WHEN (coalesce(current_setting('logidze.disabled', true), '') <> 'on') AND
  (NEW.type = 'audited_type')
EXECUTE PROCEDURE logidze_logger(<logidze params>);
```

**Reply by wiegand** [reply]
Awesome. Thank you. I will work on adding this to our codebase.
